### PR TITLE
Supermatter Cascades are not Universe-Destroying Events

### PR DESCRIPTION
**Pull request by Y0SH1M4S73R**
## About The Pull Request

Reflavors the supermatter cascade to only affect space around the station. All proc/var/type names referencing how a supermatter cascade ends the universe have been changed to reference just the concept of the supermatter cascade.

## Why It's Good For The Game

It always rubbed me the wrong way that a supermatter cascade would destroy the entire universe. 

First off, why would the Syndicate want to destroy all of reality by causing one? How would a coalition of not-quite-megacorporations whose domain is restricted to a portion of a single galaxy rationalize destroying not just their own galaxy, but the hundreds of billions of other galaxies out there. To call it cutting off one's nose to spite one's face would be the understatement of all time. 

Secondly, given the necessary conditions to cause one, and that it only causes like 5 minutes to destroy the entire universe when it happens, how inconceivably low are the odds that one hasn't happened naturally over the 14-billion-and-some-change years the universe has existed? There are probably trillions of supermatter crystals that have come into existence over the course of the universe's lifespan, across hundreds of billions of galaxies. Knowing the reactions that cause the formation of hypernob and antinob, you think it would be *that* unlikely for a *single* one of those trillions of crystals being exposed to sufficient concentrations of the requisite gasses from the nearest star?

Making the effects of a supermatter cascade non-universal provide reasonable answers to both of those questions. The Syndicate could easily rationalize consuming a large area in a cascade to get at Nanotrasen, while naturally-occuring cascades would simply leave behind colossal chunks of supermatter, which could also be the very deposits supermatter crystals and shards are initially harvested from.

## Changelog

:cl:
spellcheck: Recent re-evaluations of the underlying math have concluded that a supermatter cascade will not, in fact, destroy the entire universe - only an area several hundred kilometers in radius.
/:cl:

### CODE REVIEW

**Review by MrMelbert — COMMENTED** [review]
The universal collapse thing is so cartoonishly over the top so i'm ok with this

**Comment by Coastfront** [comment]
The tree ended its rivalry by lighting the forrest onfire is the same logic.
Also, imo this is better not just from the logic standpoint of "ah yes, surely if I blind everyone nobody will ever be able to see how ugly I am!" that the syndicate would need, which, sure, that can be aruged, but everyone is dead and blind.
But also, for the reason of. Are we really gonna say, nanotrasen, wouldn't use the fact that they can FUCKING END THE ENTIRE UNIVERSE as just an absurd counterthreat to LITERALLY ANYTHING.
its the same logic of a nuclear war, we're all in a nitroglycerin factory waving around matches and swinging bats, and its more about the satisfaction of who does it, except now on the scale of everything dying ever.

**Comment by Ghilker** [comment]
I like to think that each station is in its own pocket universe and that the whole nanotransen and the syndicates operate on multi dimensional universes and they try to annihilate each other in every way possible

File: code/controllers/subsystem/ticker.dm [hunk 1]
```diff
@@ -598,7 +598,7 @@ SUBSYSTEM_DEF(ticker)
 		if(GANG_DESTROYED)
 			news_message = "The crew of [decoded_station_name] would like to thank the Spinward Stellar Coalition Police Department for quickly resolving a minor terror threat to the station."
 		if(SUPERMATTER_CASCADE)
-			news_message = "Universal reboot initiated on [decoded_station_name] following a major supermatter cascade."
+			news_message = "Recovery of the surviving crew of [decoded_station_name] is underway following a major supermatter cascade."
```
Comment (Ghilker) [inline code/controllers/subsystem/ticker.dm:601]: This is the thing, there isn't any surviving crew, or anyone that can recover them for that matter

**Comment by Ghilker** [comment]
Also this makes an interesting round ender into a normal delamination attempt

**Comment by Acer202** [comment]
> Also this makes an interesting round ender into a normal delamination attempt

It's still an interesting round-ender. It's just that this way, people can talk about seeing this sort of event ICly without having to be a universe-hopping time lord. The Syndicate ending the entire universe is just too over-the-top and makes no sense.

**Comment by tralezab** [comment]
as an inbetween, sector destroying? it is really hard to pull this off

File: code/controllers/subsystem/ticker.dm (hunk 1 above)
Comment (tralezab) [reply code/controllers/subsystem/ticker.dm:601]: i at least agree with one thing: there is no need for recovery because everyone would 100% be dead

**Comment by Ghilker** [comment]
Questions taken from discord (my idea of lore) 
> why does the syndicate want to destroy reality

They work on multiple different universes and dimensions, they tried everything on this one and the last resort is to end it because they are on the verge of losing in there. The multiple universe is not known for the normal person, only the high command of the syndicates and of nanotransen knows about it (similar to the scp O5s)

> why does it take five minutes to end it

It doesn't, normally you'd need to create Hypernob and antinob to trigger it (and is a strongly secret way to do it), antags gets a special crystal that the syndicates created that is on perfect tune and frequency to trigger a cascade

**Comment by Redrover1760** [comment]
> They work on multiple different universes and dimensions, they tried everything on this one and the last resort is to end it because they are on the verge of losing in there. The multiple universe is not known for the normal person, only the high command of the syndicates and of nanotransen knows about it (similar to the scp O5s)
> 
> >

I'm sorry but... That's not how multiple universes work. If NT was a multi-universal corporation, than effectively in every other universe outside of the one ss13 is set in there should be different races, peoples, cultures, etc due to NT invading other universes instead of simply just new "timelines". Aka, after someone wiped out the first universe with one cascade, you are no longer allowed to play as moths/lizards/cats/etc and instead new species. (I assume some humans were transferred along the time) because fundamentally unless some incredibly deep and weird conspiracy that is incredibly hard to roleplay exists about transfering nonhumans and humans across stations fundamentally and that literally no one misses the new universes.

Also it basically makes a round truly non-canon to the universe and lore due to **destroying the universe it inhabits**. Inherently this would wipe out the moff fleets, lizard planets that exist, wherever all those other races come from, earth, etc, and replacing it without switching timelines is very difficult but at the same time by fucking with timelines you open up the ability to time travel... Not ideal.

**Comment by MetalClone** [comment]
Why can't the syndicate just open a pocket dimension to a place full of plasma and bluespace crystals with no nanotrasen there? Over reliance on alternate universes is as big as a pitfall as time travel.
My take on Super Matter Crystals is that they're made of pure neutrons, something similar to neutronium, but they're actually very stable and obviously much weaker. The reason the super matter requires such pampering, specifically through complicated atmospheric set-ups, is because its neutrons are beginning to destabilize and want to explode outward due to their unnatural form. Unlike a Neutron star, the Super Matter Crystal's synthetic origin and structure cause sit to destabilize if activated, and it requires special gaseous environments to stabilize it. When the crystal is not inert, and begins to overheat, the neutrons start to destabilize back into protons and electrons, causing it to give off immense heat and due to its astronomically minuscule size, it'll explode within minutes or seconds. 
I also like to think Super Matter 'dusts' things that interact with it, because when something makes direct contact with it, all its atoms are basically gravitated to the core of the crystal, due to its immense density, and rapidly are spread across the crystal which is what causes the 'dusting' effect.

So when the Super Matter Crystal begins to cascade effect, I like to think the uniquely unstable reaction hypernoblium and antinoblium have with eachother (I like to imagine anti-noblium reacts into some kind of anti-matter which starts to react with the super matter causing it to lose some its stable properties and which causes it to start becoming actual neutronium. This in-turn causes it neutralize its environment and everything around it by turning those particles into neutrons. Everything that touches it becomes pure neutrons, and then part of the super neutronium structure, and only stops once it has nothing left to convert because of the vaccum of space. At which it probably then turns into a giant super matter crystal and becomes inert as it rapidly gives off heat or just violently explodes into dust.

File: code/modules/antagonists/traitor/objectives/final_objective/supermatter_cascade.dm [hunk 2]
```diff
@@ -20,13 +20,13 @@
 	replace_in_name("%AREA%", initial(dest_crystal_area_pickup.name))
 	return TRUE
 
-/datum/traitor_objective/final/end_reality/generate_ui_buttons(mob/user)
+/datum/traitor_objective/final/supermatter_cascade/generate_ui_buttons(mob/user)
 	var/list/buttons = list()
 	if(!sent_crystal)
-		buttons += add_ui_button("", "Pressing this will call down a pod with the universe ending kit.", "biohazard", "destabilizing_crystal")
+		buttons += add_ui_button("", "Pressing this will call down a pod with the supermatter cascade kit.", "biohazard", "destabilizing_crystal")
```
Comment (ATH1909) [inline code/modules/antagonists/traitor/objectives/final_objective/supermatter_cascade.dm:26]: what's the "biohazard" argument here for?

File: code/modules/antagonists/traitor/objectives/final_objective/supermatter_cascade.dm [hunk 3]
```diff
@@ -20,13 +20,13 @@
...
 	return buttons
 
-/datum/traitor_objective/final/end_reality/ui_perform_action(mob/living/user, action)
+/datum/traitor_objective/final/supermatter_cascade/ui_perform_action(mob/living/user, action)
 	. = ..()
 	switch(action)
 		if("destabilizing_crystal")
```
Comment (ATH1909) [inline code/modules/antagonists/traitor/objectives/final_objective/supermatter_cascade.dm:32]: just a little below here, there's some text that calls the destabilizing crystal a "bioterror agent" that's probably from some copy+pasted code. if you could fix that, that'd be swell.

**Comment by Redrover1760** [comment]
Okay, so after learning what pocket universe interpretation stuff is. This cascade lore basically enforces an interpretation of the lore that is more meta based and ignores some pretty crucial flaws, such as:

- Backstories. Each character must have a canonical backstory. After all, watchie lived on the moffic fleet before being hired by nanotrasen, and the moffic fleet is essentially annihilated after one idiot decides to cascade. This violates the lore.
- Non-Syndi Non-NT corporations/entities (Where the fuck did space wizards, space ninjas, changelings, cults, revs?, space dragons, space carp, etc etc etc come from?) Are they just playthings of corporations now?
- This is bad writing. I understand that this is your interpretation of the lore, which is fine. However, forcing an interpretation of the lore is wrong onto other people ruins the mystery and suspense. There is nothing wrong as long as our interpretations of the lores stay both our own headcanons. You don't need to explain all the weird stuff about SS13 to try to justify limitations as intentional as in the lore. Some people would prefer if the universe was a single universe and NT and the syndicates were just another part of the corporate war instead of some kind of overpowered spacetime manipulators.
- It doesn't really improve on the lore in anyway other than forcing your interpretation to be the correct one. Nor is the lore built around a definitive interpretation. Even the little holes and the nonsense breaks this interpretation apart as intended.
- There is no reason we cannot have a compromise and just make this a sector destroying event. About as devastating without confusing people with lore and also letting the pocket dimension interpretation still stand because of course NT is lying to their employees, especially when they are all about to die. When have they not? NT's dialogue here implies that the effects of the cascade are actually touching them, instead of doing nothing outside of a pocket dimension. Why would NT show weakness like this? Maybe a hint or two that NT is suffering from the successful cascade, but actively admitting defeat is way overkill unless they were all about to die and it didn't matter anyways. **NT be like: Singularity black hole that murders everyone on station and consumes it in its entirety: I sleep. Crystal mass that consumes the entire station and murders everyone except the people who jump into a space rift thing that apparently die because no one bothered to code in some kind of more interesting ending for people who legitimately escape the crystal tide: Oh fuck Oh Fuck we are all going to die in 5 minutes.**

and most importantly, it completely fucks with backstories, confusing people wanting to write actual backstories for their characters and lore when we have to add "Oh, but everything you live in is a pocket dimension now". **Imagine writing a character for a second on MRP with a real backstory because you wanted to do one for fun, and having to deal with "But you see, I used to not live in a pocket dimension." And since I am pretty sure the real universe destroying interpretation was what cascading was built in mind for, that also means the slow moving crystal growth on the station = literally the whole universe dies, which makes 0 sense. If you wanted to signal a universe dying, you really need to amp it up with ungodly amounts of black holes or some shit. Did I mention you also just wiped out the moffic fleet and everyone of the sort? And any and all non-NT non-syndi backstory elements, of which there are quite a few?**

How do random people just enter these pocket dimensions from other corporations and space wizards and stuff. Or space dragons, as well.

File: code/modules/antagonists/traitor/objectives/final_objective/supermatter_cascade.dm (hunk 2 above)
Comment (Y0SH1M4S73R) [reply code/modules/antagonists/traitor/objectives/final_objective/supermatter_cascade.dm:26]: I think it's a reference either to the icon file or to the background to use for the button.

**Comment by githubuser4141** [comment]
> 

Is this a copypasta? Your first 2 points aren't even relevant to this PR. Changlings, wizards, etc aren't affected by this story change in any meaningful way. Your whole post is word salad and can't be made sense of. It's barely related to what this PR touches at all. The only semi-coherent point was about the moth fleet, and, well, 700 km, while not very large IRL space, is still sufficient for a _hollywood-based_ fleet to be destroyed within. In Star Wars for example, space battles take place not only within visual range, but within cannonball range ala under a 1km apart. That's plenty of room to fit in a 'fleet' of ships, considering that SS13 fleets likely consist of about 100ish spaceships in total, sure.

**Comment by Mothblocks** [comment]
why are you commenting on a pr made 10 months ago?

**Pull request merged**